File: api/schemas/dispatch.py
from typing import Any, Dict, List

from pydantic import BaseModel, Field

from api.schemas.scenario import SimulationRequest


class PricePoint(BaseModel):
    hour: int
    price: float


class DispatchRequest(BaseModel):
    scenario: SimulationRequest
    price_profile: List[PricePoint]

    low_price_threshold: float
    high_price_threshold: float
    initial_soc_kwh: float = Field(default=0.0, ge=0)


class DispatchResponse(BaseModel):
    success: bool
    results: List[Dict[str, Any]]

File: frontend/src/types/dispatch.ts
import type { SimulationRequest } from "./simulation";

export interface DispatchPricePoint {
  hour: number;
  price: number;
}

export interface DispatchRequest {
  scenario: SimulationRequest;
  price_profile: DispatchPricePoint[];

  low_price_threshold: number;
  high_price_threshold: number;
  initial_soc_kwh: number;
}

export interface DispatchResultItem {
  [key: string]: unknown;
}

export interface DispatchSummary {
  [key: string]: unknown;
}

export interface DispatchResponse {
  success: boolean;
  results: DispatchResultItem[];
  summary?: DispatchSummary;
}
